### PR TITLE
Add JSONLine parser

### PR DESCRIPTION
**Pull request by Drosaca**
it works on my machine

### CODE REVIEW

File: src/modules/Parsing/parseAsJSONL.ts [hunk 1]
```diff
@@ -10,5 +11,33 @@ export default async function parseAsJSONL<T = any>(
...
-  return Promise.reject(new Error('Not implemented'))
+  const content: Array<T> = []
+  let i = 0
+  const stream = pipeline.run()
+  const rl = readline.createInterface({
+    input: stream,
+  });
+
+  rl.on('line', (line) => {
+    if (i < (options?.pagination?.offset || 0)) {
```
Comment (super-bunny) [inline src/modules/Parsing/parseAsJSONL.ts]:
```suggestion
    if (i < (options?.pagination?.offset ?? 0)) {
```

File: src/modules/Parsing/parseAsJSONL.ts [hunk 2]
```diff
@@ -10,5 +11,33 @@ export default async function parseAsJSONL<T = any>(
...
+      return
+    }
+    content.push(JSON.parse(line))
+    i += 1
+  });
+
+  return new Promise <Array<T>>((resolve, reject) => {
+    rl.on('close', () => {
+      resolve(content)
+    })
```
Comment (super-bunny) [inline src/modules/Parsing/parseAsJSONL.ts]:
```suggestion
    rl.on('close', () => resolve(content))
```

File: src/modules/Parsing/parseAsJSONL.ts [hunk 3]
```diff
@@ -10,5 +11,33 @@ export default async function parseAsJSONL<T = any>(
...
+    stream.on('error', (e) => {
+      reject(e)
+    })
```
Comment (super-bunny) [inline src/modules/Parsing/parseAsJSONL.ts]:
```suggestion
    stream.on('error', (e) => reject(e))
```

**Review by super-bunny — DISMISSED** [review]
Hand tested on my side, seems good :slightly_smiling_face: :+1: 
Check out my code suggestions.

**Review by super-bunny — APPROVED** [review]

**Review by jess-sys — APPROVED** [review]

**Pull request merged**